packages/grid/src/Grid.tsx

```diff
@@ -1682,43 +1662,44 @@ class Grid extends PureComponent<GridProps, GridState> {
     this.startDragTimer(event);
   }
 
-  handleDoubleClick(event: MouseEvent): void {
+  handleDoubleClick(event: React.MouseEvent): void {
     this.notifyMouseHandlers('onDoubleClick', event);
   }
 
-  handleMouseMove(event: MouseEvent): void {
+  handleMouseMove(event: React.MouseEvent): void {
     this.notifyMouseHandlers('onMove', event);
   }
 
-  handleMouseLeave(event: MouseEvent): void {
+  handleMouseLeave(event: React.MouseEvent): void {
     this.notifyMouseHandlers('onLeave', event, false);
     this.setState({ mouseX: null, mouseY: null });
   }
 
   handleMouseDrag(event: MouseEvent): void {
     this.setState({ isDragging: true });
-    this.notifyMouseHandlers('onDrag', event, true, true);
+    this.notifyMouseHandlers(
+      'onDrag',
+      (event as unknown) as React.MouseEvent,
+      true,
+      true
+    );
 
     this.stopDragTimer();
   }
 
   handleMouseUp(event: MouseEvent): void {
-    window.removeEventListener(
-      'mousemove',
-      (this.handleMouseDrag as unknown) as EventListenerOrEventListenerObject,
-      true
-    );
-    window.removeEventListener(
-      'mouseup',
-      (this.handleMouseUp as unknown) as EventListenerOrEventListenerObject,
-      true
-    );
+    window.removeEventListener('mousemove', this.handleMouseDrag, true);
+    window.removeEventListener('mouseup', this.handleMouseUp, true);
 
     if (event.button != null && event.button !== 0) {
       return;
     }
 
-    this.notifyMouseHandlers('onUp', event, false);
+    this.notifyMouseHandlers(
+      'onUp',
+      (event as unknown) as React.MouseEvent,
```

Comment:
Make function param `React.MouseEvent?`

Reply:
Made a GridMouseEvent to handle these cases